activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -787,8 +787,8 @@ def supports_rename_column?
         def configure_connection
           variables = @config.fetch(:variables, {}).stringify_keys
 
-          # By default, MySQL 'where id is null' selects the last inserted id; Turn this off.
-          variables["sql_auto_is_null"] = 0
+          # Since MySQL 5.5 sql_auto_is_null is off by default, but if it is set for any reason, we need to turn it off.
+          variables["sql_auto_is_null"] = 0 unless query_value("select @@sql_auto_is_null") == 0
```

Comment:
I get that people may have flipped this config, but honestly I think it's unlikely, and for the very rare people who did that, we should probably recommend to add this to their `database.yml`.

I'd rather make it a bit harder to config for these theoretical people, than add an extra query on connect for everyone.